### PR TITLE
Update Requirements to newest version

### PR DESCRIPTION
**Pull request by plettich**
For upcoming privacyIDEA 3.6

### CODE REVIEW

**Comment by plettich** [comment]
I think we should squash the commits before merging...

**Comment by cornelinux** [comment]
> I think we should squash the commits before merging...

Oh oh. I read "I think we should smash the commits before merging..." and was a bit puzzled.

**Comment by plettich** [comment]
Working on #2570

**Comment by cornelinux** [comment]
@plettich  Is this still in Draft or ready for review?

File: privacyidea/api/auth.py [hunk 1]
```diff
@@ -365,11 +365,11 @@ def get_auth_token():
                         "authtype": authtype,
                         "exp": datetime.utcnow() + validity,
                         "rights": rights},
-                       secret, algorithm='HS256').decode('utf8')
+                       secret, algorithm='HS256')
```
Comment (cornelinux) [inline privacyidea/api/auth.py:368]: Shouldn't we move this hidden fix to an extra issue/PR?
Or is this due to a change in the requirements.

File: tests/test_lib_usercache.py [hunk 2]
```diff
@@ -105,6 +105,8 @@ def test_01_get_username_from_cache(self):
         self.assertEqual(u_name, "")
 
     def test_02_get_resolvers(self):
+        # enable user cache
```
Comment (cornelinux) [inline tests/test_lib_usercache.py:108]: I think the changes in the test is also part of a fix?
I would also like to move this to another commit/PR.

**Review by cornelinux — CHANGES_REQUESTED** [review]
I think 2 issues/PRs got mixed up here?

File: privacyidea/api/auth.py (hunk 1 above)
Comment (plettich) [reply privacyidea/api/auth.py:368]: The `jwt.encode()` returns a unicode string for Python > 3.5 so the `.decode()` fails. I have added the `to_unicode(token)` later to always return a unicode string.
This is due to an update in PyJWT.

File: tests/test_lib_usercache.py (hunk 2 above)
Comment (plettich) [reply tests/test_lib_usercache.py:108]: This is due to an update of the SQLALchemy package. The `UserCache` entry will be removed after the `self._delete_realm()` call so we can't access the data anymore.

**Review by cornelinux — APPROVED** [review]
